### DISCUSSION TITLE
Decorate NumericValidationBehavior value

### DISCUSSION THREAD

**Discussion by stephenhauck** [xamarin/XamarinCommunityToolkit#1062] [Q&A]
### Description
I am using a NumericValidationBehavior
![image](https://user-images.githubusercontent.com/12648793/110839442-8d642d00-8271-11eb-9b5b-2dbb0eea5db4.png)

And a converter to format the display ...
`  public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return Decimal.Parse(value.ToString()).ToString("C");
        }`

![image](https://user-images.githubusercontent.com/12648793/110839674-d320f580-8271-11eb-886d-dec278634030.png)

How do I format the output for the user but validate the input ?

**Comment by AndreiMisiukevich** [comment]
Hi @stephenhauck 
that's a VERY good question :)

You will have to extend NumericValidationBehavior.

```csharp
public class MoneyValidationBehavior : NumericValidationBehavior {
	protected override object? Decorate(object? value)
		=> base.Decorate(value)?.ToString()?.Replace("$", string.Empty);
}
```

And use MoneyValidationBehavior )

**Reply by stephenhauck** [reply]
Thanks, that's awesome to know !
The code you posted was not compiling so it ended up looking like this 
```csharp
protected override object DecorateValue() => base.DecorateValue()?.ToString()?.Replace("$", string.Empty);
```

**Reply by AndreiMisiukevich** [reply]
@stephenhauck oh... I guess you should use the latest pre-release version (or you will have to fix it in the future).
I made breaking change in this method. So it will be with parameter.
